docs/ml-clustering.md

```diff
@@ -0,0 +1,25 @@
+---
+layout: global
+title: Clustering - ML
+displayTitle: <a href="ml-guide.html">ML</a> - Clustering
+---
+
+In `spark.ml`, we implement the corresponding pipeline API for 
```

Comment:
nit: The pipelines API is only a subset of `mllib.clustering` (at least the documentation certainly is), so I found this language a bit confusing. Can we omit it altogether at least until we have implemented/documented more than a single feature?